tests/integration/ha/test_ha.py

```diff
@@ -117,6 +135,68 @@ async def test_replication_across_members(
     await assert_continuous_writes_consistency(c_writes)
 
 
+@pytest.mark.abort_on_fail
+async def test_kill_db_process_node_with_primary_shard(
+    ops_test: OpsTest, c_writes: ContinuousWrites, c_balanced_writes_runner
+) -> None:
+    """Check cluster can self-heal + data indexed/read when process dies on node with P_shard."""
+    app = (await app_name(ops_test)) or APP_NAME
+
+    units_ips = get_application_unit_ids_ips(ops_test, app)
+    leader_unit_ip = await get_leader_unit_ip(ops_test, app=app)
+
+    # find unit hosting the primary shard of the index "series-index"
+    shards = await get_shards_by_index(ops_test, leader_unit_ip, ContinuousWrites.INDEX_NAME)
+    first_unit_with_primary_shard = [shard.unit_id for shard in shards if shard.is_prim][0]
+
+    # Killing the only instance can be disastrous.
+    if len(ops_test.model.applications[app].units) < 2:
+        await ops_test.model.applications[app].add_unit(count=1)
+        await ops_test.model.wait_for_idle(
+            apps=[app],
+            status="active",
+            timeout=1000,
+            idle_period=IDLE_PERIOD,
+        )
+
+    # Kill the opensearch process
+    await send_kill_signal_to_process(
+        ops_test, app, first_unit_with_primary_shard, signal="SIGKILL"
+    )
+
+    # verify new writes are continuing by counting the number of writes before and after 5 seconds
+    writes = await c_writes.count()
+    time.sleep(30)  # the opensearch service is scheduled to restart after 20 sec
```

Comment:
So the idea here is that writes continue even after the primary was killed. (i.e. can the cluster support writes after the primary was killed. Hence, the amount to sleep here should be "the amount of time for opensearch to elect a new primary" **not** "the amount of time for the service to come back up.

The reasoning here is that if for some reason the old primary never came back up that the cluster would still support writes. I hope that makes sense and that I didn't butcher the explanation 😅